From 1020af779f3d3896e3fef5680b262aac525dd495 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 25 May 2022 04:05:56 +0300
Subject: [PATCH] [SW-0] rom: remove sleep in recovery indication

As the master's recovery indication is a pulse of 10us, the slave has
a limited window to capture it. Due to this, it is better to poll for
the recovery indication continuously without sleep. As the slave does
not have a parallel execution scenario, this change would not impact
functionally.

Change-Id: I3bdd0ddf0bcf71272dc6408e81e9b99695ee2e56
---

diff --git a/arch/gaudi3/arc/slave.c b/arch/gaudi3/arc/slave.c
index 5f19c32..5d3faf9 100644
--- a/arch/gaudi3/arc/slave.c
+++ b/arch/gaudi3/arc/slave.c
@@ -24,7 +24,6 @@
 	regmap_writel(val, mmD0_PSOC_GLOBAL_CONF_DIE_RECOVERY_OUT);
 }
 
-#define WAIT_SLEEP_DURATION_US		1
 void arcboot_rom_wait_for_signal(void)
 {
 	uint32_t val;
@@ -36,8 +35,6 @@
 		val = regmap_readl(mmD0_PSOC_GLOBAL_CONF_DIE_RECOVERY_IN);
 		if (val & D0_PSOC_GLOBAL_CONF_DIE_RECOVERY_IN_IND_M)
 			execute_recovery();
-
-		udelay(WAIT_SLEEP_DURATION_US);
 	}
 }
 
